docs/src/contributing/custom_frontend.md

```diff
@@ -0,0 +1,62 @@
+# Creating a Custom Frontend
+
+The backend and frontend are completely decoupled, and run in independent Docker containers. They only communicate over the [Lemmy API](websocket_http_api.md), which makes it quite easy to write alternative frontends.
+
+## Development
+
+You can use any language to create a custom frontend. The easiest option would be forking our [official frontend](https://github.com/LemmyNet/lemmy-ui), [lemmy-lite](https://github.com/IronOxidizer/lemmy-lite), or the [lemmy-frontend-example](https://github.com/LemmyNet/lemmy-front-end-example). In any case, the principle is the same: bind to `LEMMY_EXTERNAL_HOST` (default: `localhost:8536`) and handle requests using the Lemmy API at `LEMMY_INTERNAL_HOST` (default: `lemmy:8536`). Also use `LEMMY_HTTPS` to generate links with the correct protocol.
+
+To test, follow the [local development instructions](local_development.md) (you only need the database and backend dependencies). Then do `cargo run` in the Lemmy repo. Start your frontend, using `localhost:8536` for the API, and serving requests on another port. 
```

Comment:
Here I would link to docker development instead... it does run the lemmy-ui (which would be unecessary for someone developing an alternate front end), but it also sets up pictrs and iframely too, which alternate front ends might need to make sure images are being rendered.